modules/gapi/src/streaming/onevpl/accelerators/utils/elastic_barrier.hpp

```diff
@@ -0,0 +1,296 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_ACCELERATORS_UTILS_ELASTIC_BARRIER_HPP
+#define GAPI_STREAMING_ONEVPL_ACCELERATORS_UTILS_ELASTIC_BARRIER_HPP
+#include <atomic>
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+template<typename Impl>
+class elastic_barrier {
```

Comment:
Is the idea of this structure described anywhere?

Reply:
(I'd expect to see it here)

Reply:
May i add in the next PR such description?
In short overview: 
pass only single one thread (the first one) to make some initialization: resource acquisition - another will waiting for the first thread job completion.
And make final deinitialization: resource release - only for last one( if no other threads hold resource or intendent to acquire it)

Some kind of double check lock patter in Gstreamer MediaResource Adapter - but support more scenarios( and more efficient?)